nav2_behavior_tree/include/nav2_behavior_tree/bt_action_server_impl.hpp

```diff
@@ -284,6 +284,8 @@ void BtActionServer<ActionT>::executeCallback()
       action_server_->terminate_all(result);
       break;
   }
+
+  cleanErrorCodes();
```

Comment:
What about the other return above if not `on_goal_received_callback_`

Reply:
In current implementations (`NavigateToPose` and `NavigateThroughPoses`) `on_goal_received_callback_` will fail only if BT action server will fail to load behavior tree (`loadBehaviorTree(bt_xml_filename)`). If I understand correctly, `error_code` in the blackboard is being filled during BT run, rather than during XML-file load.
In other words, status clean mechanism is required to keep clean/correct errors between two sequential BT runs. And if some problem will appear on `on_goal_received_callback_` stage, we will not go through BT run in this case, but on next `BtActionServer::executeCallback()` call again will try to execute `on_goal_received_callback_` which will load/initalize BT tree from XML again.
So, this should be unnecessary in this case.

However, here is another side of view: if we won't rely on current implementations and new navigator might appear in the future (for example `NavigateAroundPoses`), and this new navigator will do something with blacklist's error codes at `goalReceived()` stage, we need to make it safe-and-sane, and clean all error status whenever it possible.

I'd rather to rely on current implementations, but if you think we need to do this anyway, please let me know, I'll give an update.

Reply:
Thinking about it twice made me decide that we'd better to have resilience to further code changes, and thus will avoid unnecessary failures. So added additional error clean in https://github.com/ros-planning/navigation2/pull/3785/commits/2912c92364a012e2da055e9878bd1053da1af4ec